inc/Engine/CDN/CDN.php

```diff
@@ -60,7 +60,7 @@ function( $matches ) {
 	 * @return string
 	 */
 	public function rewrite_srcset( $html ) {
-		$pattern = '#\s+(?:data-lazy-|data-)?srcset\s*=\s*["\']\s*(?<sources>[^"\',\s]+\.[^"\',\s]+(?:\s+\d+[wx])?(?:\s*,\s*[^"\',\s]+\.[^"\',\s]+\s+\d+[wx])*)\s*["\']#i';
+		$pattern = '#\s+(?:' . $this->get_srcset_attributes() . ')?\s*=\s*["\']\s*(?<sources>[^"\',\s]+\.[^"\',\s]+(?:\s+\d+[wx])?(?:\s*,\s*[^"\',\s]+\.[^"\',\s]+\s+\d+[wx])*)\s*["\']#i';
```

Comment:
The value returned by `get_srcset_attributes()` is the following: `data-lazy-srcset|data-srcset`

Do you see what's missing in the new pattern?

The initial pattern `(?:data-lazy-|data-)?srcset` means that:
- `data-lazy-` or `data-` are optional string values that _can_ (indicated by the `?`) be before the `srcset` string
- `srcset` is required

So this pattern matches `data-lazy-srcset`, `data-srcset` and `srcset`.

Also, the new pattern looks like so: `(?:data-lazy-srcset|data-srcset)?`
It's making the whole capturing group optional, instead of only the "suffixes" values. This could have unintended consequences.

Reply:
Thank you for the explanation @Tabrisrp.

I've made changes to the RegEx, and the `get_srcset_attributes()` method based on your comments.